components/camel-xj/src/main/java/org/apache/camel/component/xj/XmlJsonStreamWriter.java

```diff
@@ -0,0 +1,640 @@
+    public void writeCharacters(char[] text, int start, int len) throws XMLStreamException {
+        writeCharacters(new String(text, start, len));
+    }
+
+    @Override
+    public String getPrefix(String uri) throws XMLStreamException {
+        return null;
+    }
+
+    @Override
+    public void setPrefix(String prefix, String uri) throws XMLStreamException {
+    }
+
+    @Override
+    public void setDefaultNamespace(String uri) throws XMLStreamException {
+    }
+
+    @Override
+    public void setNamespaceContext(NamespaceContext context) throws XMLStreamException {
+    }
+
+    @Override
+    public NamespaceContext getNamespaceContext() {
+        throw new UnsupportedOperationException(XJConstants.UNSUPPORTED_OPERATION_EXCEPTION_MESSAGE);
+    }
+
+    @Override
+    public Object getProperty(String name) throws IllegalArgumentException {
+        throw new IllegalArgumentException(name + " unsupported");
+    }
+
+
+    private static class TreeElement {
+
+        private TreeElement parent;
+        private List<TreeElement> childs = Collections.emptyList();
+
+        private String name;
+        private String value;
+        private int xmlEvent;
+        private JsonToken jsonToken;
+
+        TreeElement(TreeElement parent, int xmlEvent, String name) {
+            this.parent = parent;
+            this.xmlEvent = xmlEvent;
+            this.name = name;
+        }
+
+        TreeElement(TreeElement parent, int xmlEvent, JsonToken jsonToken) {
+            this.parent = parent;
+            this.xmlEvent = xmlEvent;
+            this.jsonToken = jsonToken;
+        }
+
+        TreeElement(TreeElement parent, int xmlEvent, JsonToken jsonToken, String name) {
+            this.parent = parent;
+            this.xmlEvent = xmlEvent;
+            this.jsonToken = jsonToken;
+            this.name = name;
+        }
+
+        int getXmlEvent() {
+            return this.xmlEvent;
+        }
+
+        void addChild(TreeElement treeElement) {
+            if (this.childs == Collections.EMPTY_LIST) {
+                this.childs = new ArrayList<>(1);
+            }
+
+            this.childs.add(treeElement);
+        }
+
+        void setJsonToken(JsonToken jsonToken) {
+            if (this.jsonToken == null) {
+                this.jsonToken = jsonToken;
+            }
+        }
+
+        void setName(String name) {
+            this.name = name;
+        }
+
+        void setValue(String value) {
+            this.value = value;
+        }
+
+        void appendValue(String value) {
+            this.value += value;
+        }
+
+        void writeEnd() throws IOException {
+            if (jsonToken == null) {
+                writeEndNoTypeHints();
+            } else { // type hints
+                // move type hints
+                writeEndHaveTypeHints();
+            }
+        }
+
+        private void writeEndNoTypeHints() {
+            switch (xmlEvent) {
+                case XMLEvent.START_ELEMENT:
```

Comment:
This switch have just one label and default. It would be better change to if-else;

Reply:
done.